common/api.py

```diff
@@ -212,7 +212,10 @@ def __init__(self, name, path):
 
         self.name = name
         self.path = path
-        self.test_name = name.split("_")[1]
+        if name.lower().startswith('test_'):
```

Comment:
Now that determining the test name is a bit more complex,  we should consider encapsulating this parsing in a class method (e.g. ```get_test_name```)